modules/distributor/distributor.go

```diff
@@ -525,20 +554,40 @@ func requestsByTraceID(batches []*v1.ResourceSpans, userID string, spanCount int
 	return keys, traces, nil
 }
 
-func recordDiscaredSpans(err error, userID string, spanCount int) {
-	s := status.Convert(err)
-	if s == nil {
-		return
-	}
-	desc := s.Message()
+func countDiscaredSpans(responses []*tempopb.PushResponse, traces []*rebatchedTrace, repFactor int) (maxLiveDiscardedCount int, traceTooLargeDiscardedCount int) {
+	discardedTraces := make([]int, len(traces))
+	numResponses := len(responses)
+	quorum := (repFactor / 2) + 1 // min success required
 
-	if strings.HasPrefix(desc, overrides.ErrorPrefixLiveTracesExceeded) {
-		overrides.RecordDiscardedSpans(spanCount, reasonLiveTracesExceeded, userID)
-	} else if strings.HasPrefix(desc, overrides.ErrorPrefixTraceTooLarge) {
-		overrides.RecordDiscardedSpans(spanCount, reasonTraceTooLarge, userID)
-	} else {
-		overrides.RecordDiscardedSpans(spanCount, reasonInternalError, userID)
+	// max number of error allowed is {number of responses} - {minimum success required}
+	// + 1 for error count that should mark a trace as discarded
+	maxError := (numResponses - quorum) + 1
```

Comment:
i may be misunderstanding but i don't think this is true. i think there's one response for each ingester, but each trace will actually only be written to 3 (in the case of RF3). I think max error is:
```
maxError = (repFactor / 2) // integer division
```
the reason max error is below true quorum is b/c we don't require 1 write per batch to complete before this logic hits and for safety we consider it a failure.